common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -272,6 +273,36 @@ public StyledText stripAlignment() {
         });
     }
 
+    /**
+     * Cleans a StyledText into a single-line string, preserving only ascii, formatting codes, and single whitespaces.
+     * @return a new cleaned {@link StyledText}
+     */
+    public StyledText clean() {
```

Comment:
I am sorry that I still have to nag you about this PR, but I don't think this method in it's current form is suitable for `StyledText`. I think it's implementation does not fit into every other `StyledText` native method, that uses the text's parts, and does string manipulation to the parts' string itself. However, this is not my biggest concern. I think that this method does too many things, so it wouldn't be useful to other models apart from this single use case.

Here is my proposal to fix it:
1. Create a method in `StyledTextUtils` to join newlines into spaces. Basically `StyledTextUtils#joinAllLines` that is a simple call to `styledText.replaceAll("\n", " ")`. Sidenote: Do you really need to join with a space? Wouldn't that create double spaces in the place of newlines? Maybe it'd be wiser to just remove all newlines, and replace them with nothing.
2. Now, you can produce a `StyledText` without alignments, with lines joined. What does that single Unicode character look like, that you want to get rid of? If it does not change, maybe you could just call `replace(All)` to get rid of it? I understand the idea of wanting to create a method that removes all custom Unicode characters from a string, but since you don't know which characters are retextured, which are not, etc, this is not really viable at the moment.

Reply:
For 2. I made a new method that essentially remakes the StyledText in a single part. Turns out the regex doesn't work for that unicode because it's on separate parts, but I didn't want to modify the replaceAll since maybe the by-part modification was intentional? Seems like it because it mentions PartStyle